mungegithub/mungers/inactive-review-handler.go

```diff
@@ -0,0 +1,171 @@
+		Len() != 0
+}
+
+// Suggest a new reviewer who is NOT any of the existing reviewers
+// (1) get all current assignees for the PR
+// (2) get potential owners of the PR using Blunderbuss algorithm (calling getPotentialOwners() function)
+// (3) filter out current assignees from the potential owners
+// (4) if there is no any new reviewer available, the bot will encourage the PR author to ping all existing assignees
+// (5) otherwise, select a new reviewer using Blunderbuss algorithm (calling selectMultipleOwners() function with number of assignees parameter of one)
+// Note: the munger will suggest a new reviewer when the PR currently does not have any reviewer
+func (i *InactiveReviewHandler) suggestNewReviewer(issue *githubapi.Issue, potentialOwners weightMap, weightSum int64) string {
+	var newReviewer string
+
+	if len(issue.Assignees) > 0 {
+		for _, oldReviewer := range issue.Assignees {
+			login := *oldReviewer.Login
+
+			for potentialOwner := range potentialOwners {
+				if login == potentialOwner {
+					weightSum -= potentialOwners[login]
+					delete(potentialOwners, login)
+					break
+				}
+			}
+		}
+	}
+
+	if len(potentialOwners) > 0 {
+		newReviewer = selectMultipleOwners(potentialOwners, weightSum, 1)[0]
+	}
+
+	return newReviewer
+}
+
+// Munge is the workhorse encouraging PR author to assign a new reviewer
+// after getting no response from current reviewer for "COMMENTTIMECAP" duration
+// The algorithm:
+// (1) find latest comment posting time
+// (2) if the time is "COMMENTTIMECAP" or longer before today's time, create a comment
+//     encouraging the author to assign a new reviewer and unassign the old reviewer
+// (3) suggest the new reviewer using Blunderbuss algorithm, making sure the old reviewer is not suggested
+// Note: the munger will post at most "REMINDERNUMCAP" number of times
+func (i *InactiveReviewHandler) Munge(obj *github.MungeObject) {
+	issue := obj.Issue
+
+	// do not suggest new reviewer if it is not a PR, the PR has no author information, or
+	// the PR has been created more than 3 years ago (36 months with 30 days per month)
+	if !obj.IsPR() || issue.User == nil || issue.User.Login == nil ||
+		time.Since(*issue.CreatedAt) > CREATIONTIMECAP {
+		return
+	}
+
+	comments, ok := obj.ListComments()
+	if !ok {
+		return
+	}
+
+	reviewComments, ok := obj.ListReviewComments()
+	if !ok {
+		return
+	}
+
+	files, ok := obj.ListFiles()
+	if !ok || len(files) == 0 {
+		glog.Errorf("failed to detect any changed file when assigning a new reviewer for inactive PR #%v", *obj.Issue.Number)
+		return
+	}
+
+	// return if there is at least a non-author human
+	if i.haveNonAuthorHuman(issue.User.Login, comments, reviewComments) {
+		return
+	}
+
+	// return if the munger has created comments for "REMINDERNUMCAP" number of times
+	pinger := matchers.NewPinger(NOTIFNAME).SetTimePeriod(COMMENTTIMECAP).SetMaxCount(REMINDERNUMCAP)
+	if pinger.IsMaxReached(comments, issue.CreatedAt) {
```

Comment:
yup! i will create the notification with no `msg`, and put one into it if needed.